From 907b409a46af8efb63d83c46f8901d6f0d01c8fe Mon Sep 17 00:00:00 2001
From: "deepak.m1" <deepak.m1@samsung.com>
Date: Wed, 15 Apr 2015 09:27:05 -0700
Subject: [PATCH] Pressing 'Delete' key in bookmark manager's search box
 deletes bookmark.

SeparateHandler has been added to handle delete and undo commands when search box is active or focused.

BUG=462861

Review URL: https://codereview.chromium.org/1089163002

Cr-Commit-Position: refs/heads/master@{#325258}
---
 chrome/browser/resources/bookmark_manager/js/main.js | 19 +++++++++++++++----
 1 file changed, 15 insertions(+), 4 deletions(-)

diff --git a/chrome/browser/resources/bookmark_manager/js/main.js b/chrome/browser/resources/bookmark_manager/js/main.js
index acfc3d7..5bd11d1 100644
--- a/chrome/browser/resources/bookmark_manager/js/main.js
+++ b/chrome/browser/resources/bookmark_manager/js/main.js
@@ -470,6 +470,18 @@ function updatePasteCommand(opt_f) {
   });
 }
 
+function handleCanExecuteForSearchBox(e) {
+  var command = e.command;
+  switch (command.id) {
+    case 'delete-command':
+    case 'undo-command':
+      // Pass the delete and undo commands through
+      // (fixes http://crbug.com/278112).
+      e.canExecute = false;
+      break;
+  }
+}
+
 function handleCanExecuteForDocument(e) {
   var command = e.command;
   switch (command.id) {
@@ -489,10 +501,8 @@ function handleCanExecuteForDocument(e) {
       break;
 
     case 'undo-command':
-      // If the search box is active, pass the undo command through
-      // (fixes http://crbug.com/278112). Otherwise, because
-      // the global undo command has no visible UI, always enable it, and
-      // just make it a no-op if undo is not possible.
+      // Because the global undo command has no visible UI, always enable it,
+      // and just make it a no-op if undo is not possible.
       e.canExecute = e.currentTarget.activeElement !== $('term');
       break;
 
@@ -1464,6 +1474,7 @@ function continueInitializeBookmarkManager(localizedStrings) {
   });
 
   $('term').addEventListener('search', handleSearch);
+  $('term').addEventListener('canExecute', handleCanExecuteForSearchBox);
 
   $('folders-button').addEventListener('click', handleMenuButtonClicked);
   $('organize-button').addEventListener('click', handleMenuButtonClicked);
-- 
1.9.1

